R/match_hscp_lca_code.R

```diff
@@ -0,0 +1,166 @@
+        lca == "10" ~ "S37000008",
+        lca == "11" ~ "S37000009",
+        lca == "12" ~ "S37000010",
+        lca == "13" ~ "S37000011",
+        lca == "14" ~ "S37000012",
+        lca == "15" ~ "S37000013",
+        lca == "16" ~ "S37000032",
+        lca == "17" ~ "S37000015",
+        lca == "18" ~ "S37000016",
+        lca == "19" ~ "S37000017",
+        lca == "20" ~ "S37000018",
+        lca == "21" ~ "S37000019",
+        lca == "22" ~ "S37000020",
+        lca == "23" ~ "S37000021",
+        lca == "24" ~ "S37000022",
+        lca == "25" ~ "S37000033",
+        lca == "26" ~ "S37000024",
+        lca == "27" ~ "S37000026",
+        lca == "28" ~ "S37000027",
+        lca == "29" ~ "S37000028",
+        lca == "30" ~ "S37000005",
+        lca == "31" ~ "S37000030",
+        lca == "32" ~ "S37000031"
+        ),
+      # Next, use LCA to fill in ca2018
+      ca2018 = dplyr::case_when(
+        !is_missing(ca2018) ~ ca2018
+        lca == "01" ~ "S12000033",
+        lca == "02" ~ "S12000034",
+        lca == "03" ~ "S12000041",
+        lca == "04" ~ "S12000035",
+        lca == "05" ~ "S12000026",
+        lca == "06" ~ "S12000005",
+        lca == "07" ~ "S12000039",
+        lca == "08" ~ "S12000006",
+        lca == "09" ~ "S12000042",
+        lca == "10" ~ "S12000008",
+        lca == "11" ~ "S12000045",
+        lca == "12" ~ "S12000010",
+        lca == "13" ~ "S12000011",
+        lca == "14" ~ "S12000036",
+        lca == "15" ~ "S12000014",
+        lca == "16" ~ "S12000047",
+        lca == "17" ~ "S12000046",
+        lca == "18" ~ "S12000017",
+        lca == "19" ~ "S12000018",
+        lca == "20" ~ "S12000019",
+        lca == "21" ~ "S12000020",
+        lca == "22" ~ "S12000021",
+        lca == "23" ~ "S12000044",
+        lca == "24" ~ "S12000023",
+        lca == "25" ~ "S12000048",
+        lca == "26" ~ "S12000038",
+        lca == "27" ~ "S12000027",
+        lca == "28" ~ "S12000028",
+        lca == "29" ~ "S12000029",
+        lca == "30" ~ "S12000030",
+        lca == "31" ~ "S12000040",
+        lca == "32" ~ "S12000013"
+        ),
+      # Finally, use hscp2018 to fill hbrescode
+      hbrescode = dplyr::case_when(
+        !is_missing(hbrescode) ~ hbrescode,
+        hscp2019 %in% c("S37000008",
+                        "S37000020",
+                        "S37000027") ~ "S08000015",
+        hscp2019 %in% c("S37000025") ~ "S08000016",
+        hscp2019 %in% c("S37000006") ~ "S08000017",
+        hscp2019 %in% c("S37000005",
+                        "S37000013") ~ "S08000019",
+        hscp2019 %in% c("S37000001",
+                        "S37000002",
+                        "S37000019") ~ "S08000020",
+        hscp2019 %in% c(
+          "S37000009",
+          "S37000011",
+          "S37000015",
+          "S37000017",
+          "S37000024",
+          "S37000029"
+        ) ~ "S08000021",
+        hscp2019 %in% c("S37000004",
+                        "S37000016") ~ "S08000022",
```

Comment:
Should these be `hscp2018`?

See this file for the different codes which are valid for the different years [Boundary changes lookup.xlsx](https://github.com/Public-Health-Scotland/source-linkage-files/files/10268377/Boundary.changes.lookup.xlsx)

We use the 2018 boundaries as 'default' in the SLFs (so for `hbrescode`, `hbtreatcode` etc.) `lca` uses it's own set of (very old) codes, we also include the 2019 boundary variables for people to use those if they want too... 🤯

Reply:
You are right. I wrote "use 2018" in the code comments right before this piece, and then use hscp2019    >_<    Sorry